cmd/syncthing/main.go

```diff
@@ -724,7 +727,13 @@ func syncthingMain(runtimeOptions RuntimeOptions) {
 	curParts := strings.Split(Version, "-")
 	if prevParts[0] != curParts[0] {
 		if prevVersion != "" {
-			l.Infoln("Detected upgrade from", prevVersion, "to", Version)
+			// Check Syncthing version
+			checkRelation := upgrade.CompareVersions(curParts[0], prevParts[0])
+			if checkRelation == upgrade.Older || checkRelation == upgrade.MajorOlder {
```

Comment:
Inline the definition and I'd use `<`:  
`if relation := upgrade.CompareVersions(curParts[0], prevParts[0]); relation < 0`